airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobRetrier.java

```diff
@@ -51,49 +51,53 @@ public JobRetrier(JobPersistence jobPersistence, Supplier<Instant> timeSupplier)
 
   @Override
   public void run() {
-    LOGGER.info("Running job-retrier...");
+    LOGGER.info("Running Job Retrier...");
 
     final AtomicInteger failedJobs = new AtomicInteger();
     final AtomicInteger retriedJobs = new AtomicInteger();
     final List<Job> incompleteJobs = incompleteJobs();
 
-    incompleteJobs
-        .forEach(job -> {
-          if (hasReachedMaxAttempt(job)) {
-            failJob(job);
-            failedJobs.incrementAndGet();
-            return;
-          }
-
-          if (shouldRetry(job)) {
-            retriedJobs.incrementAndGet();
-            resetJob(job);
-          }
-        });
-
-    LOGGER.info("Completed Job-Retrier...");
-    LOGGER.info("Job-Retrier Summary. Incomplete jobs: {}, Job set to retry: {}, Jobs set to failed: {}",
+    incompleteJobs.forEach(job -> {
+      if (hasReachedMaxAttempt(job)) {
+        failJob(job);
+        failedJobs.incrementAndGet();
+      } else if (shouldRetry(job)) {
+        retriedJobs.incrementAndGet();
+        resetJob(job);
+      }
+    });
+
+    LOGGER.info("Completed Job Retrier...");
+    LOGGER.info("Job Retrier Summary. Incomplete jobs: {}, Job set to retry: {}, Jobs set to failed: {}",
         incompleteJobs.size(),
         failedJobs.get(),
         retriedJobs.get());
   }
 
   private List<Job> incompleteJobs() {
     try {
-      return persistence.listJobsWithStatus(JobConfig.ConfigType.SYNC, JobStatus.INCOMPLETE);
+      return persistence.listJobsWithStatus(JobStatus.INCOMPLETE);
     } catch (IOException e) {
       throw new RuntimeException("failed to fetch failed jobs", e);
     }
   }
 
   private boolean hasReachedMaxAttempt(Job job) {
-    return job.getAttemptsCount() >= MAX_ATTEMPTS;
+    if (job.getConfigType() == JobConfig.ConfigType.SYNC) {
```

Comment:
you should use the io.airbyte.scheduler.Job#REPLICATION_TYPES instead.

Reply:
done for all